internal/modx/modx.go

```diff
@@ -24,3 +25,73 @@ func Open(dir string) (m *modfile.File, err error) {
 
 	return m, nil
 }
+
+// Find locates the module version for the given import path.
+// returns zero version if none are found.
+func Find(m *modfile.File, path string) module.Version {
+	for _, pkg := range m.Replace {
+		if pkg.New.Path == path {
+			return pkg.New
+		}
+	}
+
+	for _, pkg := range m.Require {
+		if pkg.Mod.Path == path {
+			return pkg.Mod
+		}
+	}
+
+	return module.Version{}
+}
+
+// Mutate the go.mod file in the given directory.
+func Mutate(dir string, mutation func(*modfile.File) error) (err error) {
+	var mod *modfile.File
+
+	if mod, err = Open(dir); err != nil {
+		return err
+	}
+
+	if err = mutation(mod); err != nil {
+		return err
+	}
+
+	return Replace(dir, mod)
+}
+
+// Replace the go.mod file in the given directory.
+func Replace(dir string, m *modfile.File) (err error) {
+	var (
+		out  []byte
+		info os.FileInfo
+	)
+
+	m.Cleanup()
+
+	goModPath := filepath.Join(dir, "go.mod")
+
+	if out, err = m.Format(); err != nil {
+		return errors.Wrapf(err, "failed to format the 'go.mod' file: %s", goModPath)
+	}
+
+	if info, err = os.Stat(goModPath); err != nil {
+		return errors.Wrapf(err, "failed to stat the 'go.mod' file: %s", goModPath)
+	}
+
+	return errors.Wrapf(
```

Comment:
I know this is possible,

> If err is nil, Wrapf returns nil. 

It's a neat little trick. But it looks as if an error is returned here and requires investigation into the workings of `errors.Wrapf` to be sure of what happens here. Please stick with the common approach
```go
err = foobar()
if err != nil {
 return errors.Wrap(...)
}
return nil
```